app/src/test/java/homer/model/outlets/OutletTest.java

```
@@ -0,0 +1,78 @@
+        cOutlet.setValue(cOutletValue);
+        assertEquals(cOutletValue, cOutlet.getValue());
+        lOutlet.setValue(lOutletValue);
+        assertEquals(lOutletValue, lOutlet.getValue());
+
+        cOutletValue = 2.5;
+        lOutletValue = 4.0;
+
+        Throwable cOutletException = assertThrows(IllegalArgumentException.class, () -> {
+            throw new IllegalArgumentException("Value must be positive and < " + cOutlet.getMaxValue());
+        });
+
+        assertEquals("Value must be positive and < 2.0", cOutletException.getMessage());
+
+        Throwable lOutletException = assertThrows(IllegalArgumentException.class, () -> {
+            throw new IllegalArgumentException("Value must be positive and < " + lOutlet.getMaxValue());
+        });
+
+        assertEquals("Value must be positive and < 3.5", lOutletException.getMessage());
+
+        cOutletValue = -2.5;
+        lOutletValue = -4.0;
+
+        cOutletException = assertThrows(IllegalArgumentException.class, () -> {
+            throw new IllegalArgumentException("Value must be positive and < " + cOutlet.getMaxValue());
+        });
+
+        assertEquals("Value must be positive and < 2.0", cOutletException.getMessage());
+        lOutletException = assertThrows(IllegalArgumentException.class, () -> {
+            throw new IllegalArgumentException("Value must be positive and < " + lOutlet.getMaxValue());
+        });
+
+        assertEquals("Value must be positive and < 3.5", lOutletException.getMessage());
+    }
+
+    void testPlug() {
+        Light light = new Light(new DeviceInfoImpl(new DeviceIdImpl(), "LIGHT"), false);
+        final Outlet cOutlet = factory.cOutlet(new DeviceInfoImpl(new DeviceIdImpl(), "COUTLET"), 0);
+        cOutlet.plug(light);
+        assertEquals(light, cOutlet.getDevice());
+        assertEquals(light.getInfo(), cOutlet.getDevice().get().getInfo());
+        cOutlet.plug(null);
```

Won't this throw an exception?

---

The device is Optional, so passing a null device shouldn't cause problems. The test is passed anyway.

---

In the plug method I see Optional.of(...) is used, that should throw an exception?
If you want it to be empty when passing a null, Optional.ofNullable() should be used instead I think

---

The test passes because @Test is missing, beware

---

I'll check that.

---

Fixed.